src/nodes.coffee

```diff
@@ -1894,6 +1897,66 @@ exports.Obj = class Obj extends Base
       answer.push @makeCode join
     if @front then @wrapInParentheses answer else answer
 
+  # Convert "bare" properties to `ObjectProperty`s (or `Splat`s)
+  expandProperty: (property) ->
+    {variable, context, operatorToken} = property
+    key = if property instanceof Assign and context is 'object'
+      variable
+    else if property instanceof Assign
+      operatorToken.error "unexpected #{operatorToken.value}" unless @lhs
+      variable
+    else
+      property
+    if key instanceof Value and key.hasProperties()
+      key.error 'invalid object key' unless context isnt 'object' and key.this
+      if property instanceof Assign
+        return new ObjectProperty fromAssign: property
```

Comment:
Ok I think I came up with a lightweight way to "normalize" this

Now `Obj::expandProperty()` returns either an `ObjectProperty` (new node class) or a `Splat` (mimicking the target AST shape, where `ObjectProperty`s live side-by-side with `SpreadElement`s)

Ultimately we might want to extend this back through eg the grammar and use `ObjectProperty`s during compile-to-JS, but for now the conversion to `ObjectProperty`s is encapsulated here in `Obj::expandProperty()` so only used when generating AST (avoiding potential breakage of existing code/messier refactor)